cf_units/__init__.py

```diff
@@ -552,15 +552,13 @@ def _num2date_to_nearest_second(
     # later versions, if one wished to do so for the sake of consistency.
     cftime_unit = unit.cftime_unit
     time_units = cftime_unit.split(" ")[0]
-    has_half_seconds = np.logical_and(
-        time_units == "seconds", time_values % 1.0 == 0.5
-    )
-    num2date_kwargs = dict(
-        units=cftime_unit,
-        calendar=unit.calendar,
-        only_use_cftime_datetimes=only_use_cftime_datetimes,
-        only_use_python_datetimes=only_use_python_datetimes,
-    )
+    has_half_seconds = np.logical_and(time_units == "seconds", time_values % 1.0 == 0.5)
```

Comment:
Seems like ruff is overriding black here.  Maybe we shouldn't use both?

https://github.com/SciTools/cf-units/blob/cf4d66d002dc1028d33c7d80a66396a1856de6f4/pyproject.toml#L48-L49

Reply:
Actually it looks like a lot of the changes in the PR are just increasing the allowed line length.  Maybe it would be better to tell ruff to keep to 79 characters.  I've no objection in general to an 88 character limit, but it might be good to separate concerns of tool choices vs style choices.  Line limit could be increased as a follow-up PR.

Reply:
Looks like you just need a couple of lines in the `pyproject.toml` to reduce the character limit

```
[tool.ruff]

line-length = 79
```
I'd be interested to see how much smaller that makes the diff, and it might help inform the discussion at https://github.com/SciTools/iris/discussions/5254.  Looks like a lot of the other changes are introducing f-strings to replace old-style string formatting, which seems fair enough.